src/app/admin/time-table/time-table.component.ts

```diff
@@ -110,82 +114,109 @@ export class TimeTableComponent implements OnInit {
-    this.apiService.getEntityByAction('timeTable', 'getTimeTablesForSubject', subjectId)
-      .subscribe((result: any) => {
-        if (!result.response) {
-          table = result;
-          const ids = table.map(a => Number(a.group_id));
-          this.apiService.getByEntityManager('Group', ids).subscribe((value1: Group[]) => {
-            table.map(a => {
-              value1.find(obj => {
-                if (obj.group_id === a.group_id) {
-                  a.group_name = obj.group_name;
-                }
-              });
-            });
-          });
-          this.timeTable = table;
-          this.dataSource.data = table;
-          this.dataSource.paginator = this.paginator;
+  private getTimeTable(subjectId): Subscription {
+    return this.apiService.getEntityByAction('timeTable', 'getTimeTablesForSubject', subjectId).pipe(
+      switchMap(data => {
+        if (!data.response) {
+          return this.getTimeTableInfo(data);
         } else {
-          this.timeTable = [];
-          this.dataSource.data = [];
+          this.isLoaded = true;
+          return EMPTY;
         }
-      });
+      })
+    ).subscribe((data: any) => {
+        this.timeTable.push(data);
+        this.dataSource.data.push(data);
+        this.dataSource.paginator = this.paginator;
+      }
+    );
   }
 
-  private addTimeTable(data: TimeTable) {
-    this.apiService.createEntity('TimeTable', data).subscribe((result: TimeTable[]) => {
-      if (result[0].subject_id === this.subjectId) {
-        const updatedTable: TimeTable[] = result;
-        this.apiService.getEntity('Group', result[0].group_id).subscribe((value: Group[]) => {
-          updatedTable[0].group_name = value[0].group_name;
-          this.dataSource.data.push(updatedTable[0]);
-          this.table.renderRows();
-          this.dataSource.paginator = this.paginator;
+  private getTimeTableInfo(data) {
+    const table = data;
+    const ids = table.map(a => Number(a.group_id));
+    return this.apiService.getByEntityManager('Group', ids).pipe(
+      switchMap(value => {
+        table.map(a => {
+          value.find(obj => {
+            if (obj.group_id === a.group_id) {
+              a.group_name = obj.group_name;
+            }
+          });
         });
-      }
-    }, (error: any) => {
-      if (error.error.response.includes('Wrong input')) {
-        this.modalService.openInfoModal('Не правильно введені дані');
-      } else if (error.error.response.includes('ERROR: SQLSTATE[23000]: Integrity constraint violation')) {
-        this.modalService.openInfoModal('Розклад для вибраної групи та предмету вже заданий');
-      } else {
-        this.modalService.openInfoModal('Помилка оновлення');
-      }
-    });
+        return table;
+      })
+    );
+  }
+
+  private addTimeTable(data: TimeTable): Subscription {
+    let updatedTable: TimeTable[];
+    return this.apiService.createEntity('TimeTable', data).pipe(
+      catchError(err => {
```

Comment:
There is no reason to throw error here if additional logic for error was not implemented